hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -0,0 +1,694 @@
+              // Volume doesn't exist. Create it
+              try {
+                objectStore.createVolume(volumeStr);
+              } catch (OMException newVolEx) {
+                // Ignore the case where another client created the volume
+                if (!newVolEx.getResult().equals(VOLUME_ALREADY_EXISTS)) {
+                  throw newVolEx;
+                }
+              }
+            } else {
+              throw getVolEx;
+            }
+            // Try get volume again
+            volume = proxy.getVolumeDetails(volumeStr);
+          }
+          // Create the bucket
+          try {
+            volume.createBucket(bucketStr);
+          } catch (OMException newBucEx) {
+            // Ignore the case where another client created the bucket
+            if (!newBucEx.getResult().equals(BUCKET_ALREADY_EXISTS)) {
+              throw newBucEx;
+            }
+          }
+        }
+        // Try get bucket again
+        bucket = proxy.getBucketDetails(volumeStr, bucketStr);
+      } else {
+        throw ex;
+      }
+    }
+
+    return bucket;
+  }
+
+  @Override
+  public void close() throws IOException {
+    ozoneClient.close();
+  }
+
+  @Override
+  public InputStream readFile(String pathStr) throws IOException {
+    incrementCounter(Statistic.OBJECTS_READ);
+    OFSPath ofsPath = new OFSPath(pathStr);
+    String key = ofsPath.getKeyName();
+    try {
+      OzoneBucket bucket = getBucket(ofsPath, false);
+      return bucket.readFile(key).getInputStream();
+    } catch (OMException ex) {
+      if (ex.getResult() == OMException.ResultCodes.FILE_NOT_FOUND
+          || ex.getResult() == OMException.ResultCodes.NOT_A_FILE) {
+        throw new FileNotFoundException(
+            ex.getResult().name() + ": " + ex.getMessage());
+      } else {
+        throw ex;
+      }
+    }
+  }
+
+  protected void incrementCounter(Statistic objectsRead) {
+    //noop: Use OzoneClientAdapterImpl which supports statistics.
+  }
+
+  @Override
+  public OzoneFSOutputStream createFile(String pathStr, boolean overWrite,
+      boolean recursive) throws IOException {
+    incrementCounter(Statistic.OBJECTS_CREATED);
+    OFSPath ofsPath = new OFSPath(pathStr);
+    String key = ofsPath.getKeyName();
+    try {
+      OzoneBucket bucket = getBucket(ofsPath, false);
+      OzoneOutputStream ozoneOutputStream = bucket.createFile(
+          key, 0, replicationType, replicationFactor, overWrite, recursive);
+      return new OzoneFSOutputStream(ozoneOutputStream.getOutputStream());
+    } catch (OMException ex) {
+      if (ex.getResult() == OMException.ResultCodes.FILE_ALREADY_EXISTS
+          || ex.getResult() == OMException.ResultCodes.NOT_A_FILE) {
+        throw new FileAlreadyExistsException(
+            ex.getResult().name() + ": " + ex.getMessage());
+      } else {
+        throw ex;
+      }
+    }
+  }
+
+  @Override
+  public void renamePath(String path, String newPath) throws IOException {
```

Comment:
Let's change renamePath to rename

Reply:
Done in 5c1c08178a4d230ded059c8f11e82f19a375a862